overlore/townhall/service.py

```diff
@@ -20,40 +23,35 @@
 TORII_WS = os.environ.get("DEV_TORII_WS")
 TORII_GRAPHQL = os.environ.get("DEV_TORII_GRAPHQL")
 
-
-def simple_callback(argument):
-    print(argument)
-
-
-parser = argparse.ArgumentParser(description="The weaving loomer of all possible actual experiential occasions.")
-parser.add_argument(
-    "--mock",
-    action="store_true",
-    help="Use mock data for GPT response instead of querying the API. (saves API calls)",
-)
-parser.add_argument("-a", "--address", help="Host address for connection", type=str, default="localhost")
-
-args = parser.parse_args()
+args = parse_cli_args()
 
 
 def handle_sigint(a, b):
     exit(0)
 
 
-async def service(websocket: WebSocketServerProtocol, path: str):
+async def service(websocket: WebSocketServerProtocol, extra_argument):
     async for message in websocket:
-        response = await gen_townhall(message, args.mock)
+        response = await gen_townhall(message, extra_argument.get("mock"))
         await websocket.send(json.dumps(response))
 
 
 async def start():
+    db = DatabaseHandler.instance().init()
     signal.signal(signalnum=signal.SIGINT, handler=handle_sigint)
     gpt_interface = GptInterface.instance()
     gpt_interface.init(OPENAI_API_KEY, OPENAI_EMBEDDINGS_API_KEY)
-    overlore_pulse = serve(service, args.address, SERVICE_WS_PORT)
+
+    bound_handler = functools.partial(service, extra_argument={"mock": args.mock})
+    overlore_pulse = serve(bound_handler, args.address, SERVICE_WS_PORT)
+
     print(f"great job, starting this service on port {SERVICE_WS_PORT}. everything is perfect from now on.")
     # arbitrarily choose just the first realm to sub to events on
-    await asyncio.gather(overlore_pulse, torii_event_sub(TORII_WS, simple_callback))
+    await asyncio.gather(
+        overlore_pulse,
+        torii_event_sub(TORII_WS, db.process_event, Subscriptions.COMBAT_OUTCOME_EVENT_EMITTED),
```

Comment:
we have two subscriptions for two events that call into the same `process_event` method--why not have one subscription for all events that get processed.

Reply:
Couldn't get it to work by passing both event keys to the subscription query, I'll try it some more

Reply:
why filter the keys at all when the same callback is used?

Reply:
We could try but I thought it was cleaner that way. The cleanest would be to subscribe once for 2 events, I'll try that

Reply:
Not possible to subscribe with only one call such as this one: 
```
torii_event_sub(TORII_WS, db.process_event, [Subscriptions.COMBAT_OUTCOME_EVENT_EMITTED, Subscriptions.ORDER_ACCEPTED_EVENT_EMITTED]),
``` 
as subscribing to two events: 
```
subscription {
    eventEmitted(keys: ["0x20e86edfa14c93309aa6559742e993d42d48507f3bf654a12d77a54f10f8945", "0x1736c207163ad481e2a196c0fb6394f90c66c2e2b52e0c03d4a077ac6cea918"]) {
        id
        keys
        data
        createdAt
    }
}
```
does not work

Reply:
Not sure which is cleaner, subscribing to all events and just filtering or subscribing to two events

Reply:
think about it in terms of resources

per subscription, what's the cost?  per invocation of a function, what's the cost?